src/hooks/useNavigator.ts

```diff
@@ -2,32 +2,35 @@ import { ExitFormDialogContext } from "@dashboard/components/Form/ExitFormDialog
 import { useContext } from "react";
 import useRouter from "use-react-router";
 
-export type UseNavigatorResult = (
-  url: string,
-  opts?: {
-    replace?: boolean;
-    preserveQs?: boolean;
-    resetScroll?: boolean;
-  },
-) => void;
+export type NavigatorOpts = {
+  replace?: boolean;
+  preserveQs?: boolean;
+  resetScroll?: boolean;
+  state?: Record<string, unknown>;
```

Comment:
question: can't we take this props type from react-router?

Reply:
`state` has `unknown` type